exo-add/helpers/helpers.go

```diff
@@ -0,0 +1,107 @@
+package helpers
+
+import (
+	"bufio"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path"
+
+	"github.com/Originate/exosphere/exo-add/os_helpers"
+	"github.com/Originate/exosphere/exo-add/types"
+	"github.com/Originate/exosphere/exo-add/user_input"
+	"github.com/tmrts/boilr/pkg/template"
+	"gopkg.in/yaml.v2"
+)
+
+func CheckForService(serviceRole string, existingServices []string) {
+	if contains(existingServices, serviceRole) {
+		fmt.Printf(`Service %v already exists in this application`, serviceRole)
+		os.Exit(1)
+	}
+}
+
+func GetExistingServices(services types.Services) []string {
+	existingServices := []string{}
+	for service, _ := range services.Private {
+		existingServices = append(existingServices, service)
+	}
+	for service, _ := range services.Public {
+		existingServices = append(existingServices, service)
+	}
+	return existingServices
+}
+
+func GetTemplates() []string {
+	templatesDir := ".exosphere"
+	if !osHelpers.DirectoryExists(templatesDir) || osHelpers.IsEmpty(templatesDir) {
+		fmt.Print("no templates found")
```

Comment:
I did that but the step `And I see the error:` timed out  because`expectedText` didn't show up in `out.String()`. This is odd since `log.Fatal` should call `Print()` before `os.Exit(1)`. I have not figured out why that is so now I just call `Print` and `os.Exit(1)` separately.